File: integrations/langchain/examples/agentDidLangChain.example.js
const { ethers } = require("ethers");
const { createAgent } = require("langchain");
const { AgentIdentity } = require("@agent-did/sdk");
const { createAgentDidIntegration } = require("../src");

async function main() {
  try {
    const privateKey = process.env.CREATOR_PRIVATE_KEY;

    if (!privateKey) {
      throw new Error("Missing CREATOR_PRIVATE_KEY environment variable");
    }

    const signer = new ethers.Wallet(privateKey);
    const identity = new AgentIdentity({ signer, network: "polygon" });

    const runtimeIdentity = await identity.create({
      name: "research_assistant",
      description: "Agente de investigacion con trazabilidad Agent-DID",
      coreModel: "gpt-4.1-mini",
      systemPrompt: "Eres un investigador preciso, trazable y seguro.",
      capabilities: ["research:web", "report:write", "http:sign"],
    });

    const integration = createAgentDidIntegration({
      agentIdentity: identity,
      runtimeIdentity,
      additionalSystemContext: "Solo usa la firma HTTP Agent-DID cuando el flujo realmente requiera autenticacion saliente.",
      expose: {
        signHttp: true,
        verifySignatures: true,
        documentHistory: true,
      },
    });

    const agent = createAgent({
      name: "research_assistant",
      model: "openai:gpt-4.1-mini",
      systemPrompt: "Eres un asistente empresarial preciso. Usa herramientas cuando mejoren exactitud o trazabilidad.",
      tools: integration.tools,
      middleware: [integration.middleware],
    });

    const result = await agent.invoke({
      messages: [
        {
          role: "user",
          content: "Indica tu identidad Agent-DID actual y prepara una firma HTTP POST para https://api.example.com/orders con body {\"orderId\":42}",
        },
      ],
    });

    console.log(result.messages.at(-1)?.content);
  } catch (error) {
    console.error(error);
    process.exit(1);
  }
}

void main();

File: integrations/langchain-python/examples/agent_did_langchain_example.py
"""Minimal Agent-DID + LangChain Python assembly example."""

from __future__ import annotations

import asyncio
import os

from agent_did_sdk import AgentIdentity, AgentIdentityConfig, CreateAgentParams
from langchain.agents import create_agent

from agent_did_langchain import create_agent_did_langchain_integration


async def main() -> None:
    signer_address = os.environ.get("AGENT_DID_SIGNER_ADDRESS", "0x8888888888888888888888888888888888888888")
    model = os.environ.get("LANGCHAIN_MODEL", "openai:gpt-4.1-mini")
    identity = AgentIdentity(AgentIdentityConfig(signer_address=signer_address, network="polygon"))

    runtime_identity = await identity.create(
        CreateAgentParams(
            name="research_assistant",
            description="Agente de investigacion con identidad verificable",
            core_model="gpt-4.1-mini",
            system_prompt="Eres un agente de investigacion preciso y trazable.",
            capabilities=["research:web", "report:write"],
        )
    )

    integration = create_agent_did_langchain_integration(
        agent_identity=identity,
        runtime_identity=runtime_identity,
        expose={"sign_http": True},
        additional_system_context="Never sign arbitrary payloads.",
    )

    agent = create_agent(
        model=model,
        tools=integration.tools,
        system_prompt=integration.compose_system_prompt(
            "Responde con precision y usa herramientas solo cuando aporten evidencia verificable."
        ),
    )

    result = await agent.ainvoke(
        {
            "messages": [
                {
                    "role": "user",
                    "content": "Muestrame tu DID actual y explica cuando deberias usar la firma HTTP.",
                }
            ]
        }
    )
    print(result)


if __name__ == "__main__":
    asyncio.run(main())
